client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -901,12 +913,19 @@ const withIsLargeCurrency = ( Component: LocalizedComponent< typeof PlanFeatures
 /* eslint-disable wpcalypso/redux-no-bound-selectors */
 const ConnectedPlanFeatures2023Grid = connect(
 	( state: IAppState, ownProps: PlanFeatures2023GridProps ) => {
-		const { placeholder, plans, isLandingPage, visiblePlans, isInSignup, siteId } = ownProps;
+		const {
+			placeholder,
+			plans,
+			isLandingPage,
+			visiblePlans,
+			isInSignup,
+			siteId,
+			currentSitePlanSlug,
+		} = ownProps;
 		const canUserPurchasePlan =
 			! isCurrentPlanPaid( state, siteId ) || isCurrentUserCurrentPlanOwner( state, siteId );
```

Comment:
I'm wondering if we can replace `isCurrentPlanPaid( state, siteId )` also with something like `! isFreePlan( currentSitePlanSlug )`. This will help us remove another selector reference.